example/example2.c

```diff
@@ -139,41 +139,106 @@ static BOOL RegisterBasicAUMID(LPCWSTR aumid, LPCWSTR displayName, LPCWSTR iconU
     return TRUE;
 }
 
-int main(int argc, char ** argv)
+// Déclarer l'événement global
+HANDLE g_hEvent = NULL;
+
+// Callback functions
+void toastActivated(void *userData) {
+    printf("Toast activated without action.\n");
+    // Signaler que le callback a été appelé
+    if (g_hEvent != NULL) {
+        SetEvent(g_hEvent);
+    }
+}
+
+void toastActivatedAction(void *userData, int actionIndex) {
+    if (actionIndex == 0) {
+        printf("Button 1 clicked.\n");
+    } else if (actionIndex == 1) {
+        printf("Button 2 clicked.\n");
+    } else {
+        printf("Unknown action clicked.\n");
+    }
+    // Signaler que le callback a été appelé
+    if (g_hEvent != NULL) {
+        SetEvent(g_hEvent);
+    }
+}
+
+void toastDismissed(void *userData, WTLC_DismissalReason state) {
+    switch (state) {
+        case WTLC_DismissalReason_UserCanceled:
+            printf("Toast dismissed by user.\n");
+            break;
+        case WTLC_DismissalReason_ApplicationHidden:
+            printf("Toast dismissed by application.\n");
+            break;
+        case WTLC_DismissalReason_TimedOut:
+            printf("Toast timed out.\n");
+            break;
+        default:
+            printf("Toast dismissed with unknown reason.\n");
+            break;
+    }
+    // Signaler que le callback a été appelé
+    if (g_hEvent != NULL) {
+        SetEvent(g_hEvent);
+    }
+}
+
+void toastFailed(void *userData) {
+    printf("Toast failed to display.\n");
+    // Signaler que le callback a été appelé
+    if (g_hEvent != NULL) {
+        SetEvent(g_hEvent);
+    }
+}
+
+INT WINAPI wWinMain()
 {
-    WTLC_Instance * instance = NULL;
-    WTLC_Template * templ = NULL;
+    AllocConsole();
+    FILE* pCout;
```

Comment:
Intermingled declarations and code is C99 feature and can't be built with older MSVC compilers. Example should support widest possible number of configurations to get started with any compiler easily